modules/apps/jax-rs/mdm-admin/src/main/java/org/wso2/carbon/mdm/api/User.java

```diff
@@ -615,4 +615,33 @@ public Response updateRoles(@PathParam("username") String username, List<String>
 		}
 		return Response.status(HttpStatus.SC_OK).build();
 	}
+	/**
+	 * Get the list of usernames in the system.
+	 *
+	 * @return A list of usernames
+	 * @throws MDMAPIException
+	 */
+	@GET
+	@Path("users-by-username")
+	public Response getFilteredUsersByUsername(@QueryParam("q") String username) throws MDMAPIException {
+		log.error("HIT with param");
+		if (log.isDebugEnabled()) {
+			log.debug("Getting the list of users by name");
```

Comment:
The behaviour of this method is to retrieve the users list with the filtering option for the user name.
We don't retrieve data related to a particular user here. We just filter the list of users whom the username contains the given characters.
In this case will it be valuable to print the search query?